groups/models.py

```diff
@@ -13,6 +13,13 @@ class CommentManager(models.Manager):
     def for_discussion_pk(self, pk):
         return self.get_queryset().filter(discussion_id=pk)
 
+    def with_user_may_delete(self, user):
+        queryset = self.get_queryset()
+        for comment in queryset:
+            comment.user_may_delete = comment.may_be_deleted(user)
+
+        return queryset
```

Comment:
Yes, it might not be possible to reuse `Comment.may_be_deleted`. I'm a bit worried that if we don't use annotation here the `user_may_delete` attribute could be lost by further manipulation of the `queryset`.